### PR TITLE
Adds no-slip jackboots for Nuke Ops

### PR DESCRIPTION
**Pull request by Robustin**
E-Shield = 20TC
Belt full of bulldog ammunition = 8TC
Emag = 6TC

Getting destroyed by an assistant who watered up a pest can = Priceless

Seriously, with all the efforts to make NukeOps "fluke-proof" (Syndicate pins, Syndieborgs, etc.) why in the world would anyone take away their noslips and give them less defense against slip memes than the average traitor? 

I don't know why...  but to cut off any argument about it being non-thematic or something, I made them into jackboots unique to Nuclear mode. 

-- THESE JACKBOOTS COST 2TC AND FUNCTION SIMILAR TO REGULAR NOSLIPS --

### CODE REVIEW

**Comment by Xhuis** [comment]
i ded

**Comment by KorPhaeron** [comment]
Or you can buy noslips

**Comment by Bawhoppen** [comment]
Yea the whole point is that if you don't want to slip, you BUY them. 4 tc is very little.

**Comment by as334** [comment]
:-1: power creep

**Comment by lzimann** [comment]
Last time I played nuke ops, you couldn't buy no-slips. Was that temporary or just a bug?

**Comment by Robustin** [comment]
Is this the part where I explain I didn't get slipped and just made this PR based on OOC comments? 

Brown shoe noslips are excluded from NukeOps and the Tactical noslips didn't have an item reference, and I'm 80% sure they don't exist as a result.

/datum/uplink_item/stealthy_tools/syndigaloshes
	name = "No-Slip Brown Shoes"
	desc = "These shoes will allow the wearer to run on wet floors and slippery objects without falling down. They do not work on heavily lubricated surfaces."
	item = /obj/item/clothing/shoes/sneakers/syndigaloshes
	cost = 2
	excludefrom = list(/datum/game_mode/nuclear)

**Comment by daman997** [comment]
why the eshield price increase

**Comment by Robustin** [comment]
Oh I was just going off memory, this code doesn't touch eshields

**Comment by KorPhaeron** [comment]
In that case maybe they should just walk.

Nuke ops already have magic guns the crew cant use, now you're taking away another way for crew to fight them.

We don't need it to be some ops vs sec counterstrike knock off mode more than it already is.

**Comment by Bawhoppen** [comment]
Yea. This is kind of dumb imo.

**Comment by Robustin** [comment]
The nature of slipping is just terrible to go up against. When you're an operative your eyes already have to be in 5 places at once, not noticing that the tiles are slightly darker when the floor may already be caked in blood, bodies, and bullets, it not really a fair demand for the Ops.

Most people will still forgo them, slip memes will never die, but for combat ops who actually want to be slip resistant they should have the option to spend TC on it. They are also impractical for StealthOps.

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 1]
```diff
@@ -16,6 +16,17 @@
 	flags = NOSLIP
 	origin_tech = "syndicate=3"
 	burn_state = -1 //Won't burn in fires
+	
+/obj/item/clothing/shoes/nukegaloshes
+	desc = "Advanced combat boots utilizing proprietary 'No-Slip' technology."
+	name = "adhesion combat boots"
+	icon_state = "jackboots"
+	item_state = "jackboots"
+	armor = list(melee = 25, bullet = 25, laser = 25, energy = 25, bomb = 50, bio = 10, rad = 0)
```
Comment (GunHog) [inline code/modules/clothing/shoes/miscellaneous.dm]: Does shoe armor even do anything?

Comment (RemieRichards) [inline code/modules/clothing/shoes/miscellaneous.dm]: Nope, ```check_armor()``` never asks for shoes.

Comment (blobbernaut) [inline code/modules/clothing/shoes/miscellaneous.dm]: actually in the PR directly above this one the checkarmor is changed to include the back slot for cloaks, all the things that can have armor are shown there

**Comment by PKPenguin321** [comment]
this is actually agreeable
noslips are for getting your maintenance murderbone on as a traitor, not for ops to drop 20 TCs on so everybody can have a pair

**Comment by KorPhaeron** [comment]
Remember when nuke ops had space suits and revolvers instead of being highly kitted out tactical murder squads with the ability to buy immunity to stuns, slipping, guns, etc?

**Comment by PKPenguin321** [comment]
kor, that has nothing to do with this tbh

**Comment by Robustin** [comment]
Yes, and I recall they had access to noslip back in the revolver and spessuit days too. 

4TC or 400TC, unless someone can screenshot a picture of noslips in an operative uplink... they simply aren't there in regular or tactical form. 

The amount of confusion as to whether Ops even have access to noslips is kind of astounding. 

I'm not a code expert but the "Tactical No-Slips" I removed didn't even reference a shoe that existed, it was just an item, description, and TC cost... which might explain why Syndies don't have access to them.

**Comment by bustygoku** [comment]
Just give ops the same noslips traitors get for 2tc. It's not that big of a deal.

**Comment by GunHog** [comment]
Jackboots are more Tacticool.

**Comment by Dorsisdwarf** [comment]
Rather than give nukers noslips 4free

Why not just fix my shitty typo?

**Comment by Shapsy** [comment]
Why don't you just get one of your other op buddies to shoot the slipper and then shake you back up? Getting slipped is hardly ggnore when you ideally have four other dudes with guns right behind you

**Comment by KorPhaeron** [comment]
Or use adrenals.

Or don't step on the water.

**Comment by Iamgoofball** [comment]
# hey assholes how about we have @MrStonedOne test run this
# hey assholes how about we have @MrStonedOne test run this
# hey assholes how about we have @MrStonedOne test run this
# hey assholes how about we have @MrStonedOne test run this

**Comment by KorPhaeron** [comment]
Because it's a fucking tiny niche situation when the janitor fights an op. It comes up rarely in a single round type.

How are we going to run a meaningful test about that? How are we gonna compare nuke now vs nuke after the change in any quantifiable way, especialy with stat tracking being broken.

**Comment by PKPenguin321** [comment]
>tiny niche situation

slips are standard issue for any op fighters,

**Comment by KorPhaeron** [comment]
I'm willing to bet the average number of ops killed by slips per nuke round is somewhere less than 1. I see more ops blown up by friendly bombs than slips.

**Comment by Zilenan91** [comment]
Isn't it more than a little retarded how less than a tenth of an inch of water on the floor is more deadly than laser guns.

**Comment by Nanotransistor** [comment]
This is good. Specially since it's something less for the ops to buy. Right now, they have to waste so much time to choose the best items to buy, only to end up choosing poorly anyways. Ops should win or lose based on skill and strategy, not on annoying shopping.

There are also hundreds of ways to fight ops while not being sec. Water is not necessary and it makes for boring rounds.

**Comment by Lumipharon** [comment]
Noslips got removed from nuke ops (relatively, not sure exactly) recently, for unknown reasons.
I don't usually buy noslips, but I see zero logic in preventing ops from buyin them if they so wish.

As for the pr itself, really it should just enable noslips for nukes again, and someone should make noslips chameleon so they get meta'd less. (Or lets be honest, for fashion)

**Comment by Robustin** [comment]
@KorPhaeron 

The number of deaths from almost any source is going to be less than 1 for nukeops. 

Sometimes they detonate their bomb implant by accident, sometimes they suffocate after forgetting to turn on their airtank when they leave the shuttle, sometimes a secborg gets them, sometimes one gets stunprodded, sometimes the bartender nabs one, etc, etc. 

That's absolutely no reason for them getting the same level of slip insurance that regular traitors get. 

I mean you're talking about how marginal slips are, and I think everyone here knows that these boots won't even be purchased most of the time, so why deprive the people who do want noslip protection from that meaningful choice? 

And like I said before, it's silly to even pretend most slips are preventable. You're running down a hallway, plagued by lag, at a blazing 8 FPS, and you've got less than half a second to react to a single tile that's a couple shades darker. Plus Ops are managing a lot more on their screen than average joe taking a leisurely walk down a hall. 

Nukeops have anti-EVERYTHING, but its balanced because of the limited resources and coordination required for success. Depriving them of noslips is just irrational.

**Comment by Dorsisdwarf** [comment]
Thats why their anti-things cost TC, and why you should fix nukeslips instead of giving them supernoslips for free

**Comment by Lumipharon** [comment]
That's why you should read what the pr actually does @Dorsisdwarf

**Comment by Steelpoint** [comment]
Is almost everyone in this PR retarded?

This PR literally just lets Ops buy no slips. 

Ops have always been able to buy no slips for years!

**Comment by Metacide** [comment]
So this makes noslips buyable again, as well as halving the cost from 4TC each to 2TC?

I certainly think they should be able to buy them - if they previously couldn't that's pretty shabby. Price reduction sounds a bit powercreepy, but I suppose I don't really play as ops enough to be able to judge. Still, the crew don't have an awful lot of defenses against ops and they usually fail because of their own shoddiness rather than a super effective crew - so I hope the TC cost of removing another way to fight ops is not insignificant.

https://github.com/tgstation/-tg-station/pull/12619 should make slipping ops a little harder too, anyway.

**Comment by Bawhoppen** [comment]
Please give them just normal noslip shoes instead of combat boots.

**Comment by Steelpoint** [comment]
The Jackboots are more thematic than anything.

Also I do agree that the crew themselves have little in the way to actually fight Ops, it shows how poor things are when a bottle of water is more efficient than anything Sec has to offer.

But nonetheless the removal of the Op no slips was stupid and likely due to a mistake.

This PR is correcting that mistake.

**Comment by Bawhoppen** [comment]
Jackboots are not more thematic when you're doing stealth.

**Comment by KorPhaeron** [comment]
Also I really dont have the energy for another fight about armour and it probably wont matter since its the legs but since armour straight up stacks it'd be kind of weird that special shoes would almost double the hardsuit protection

**Comment by Robustin** [comment]
It's a meme at this point that coders barely understand how the game works, but I thought you guys would at least know your own code.

The armor values are what I straight copy-pasted from combat boots for NukeOps, like scroll down 4 lines and you'll see what I mean. Plus the previous conversation here makes it seem like that armor effect is marginal at best.

**Comment by Dorsisdwarf** [comment]
Coders generally only understand code they've previously worked with, or, in my case, dont understand code at all.

also, :+1:

**Comment by KorPhaeron** [comment]
Yeah sorry for not knowing all 100k lines of code by heart.

And only recently has someone made a PR for combat boots to get counted in calculations at all.

**Comment by Zilenan91** [comment]
I don't think boots actually give armor to anything at all. I could be wrong, but I know that jackboots have armor values that don't actually do anything.

**Comment by MrStonedOne** [comment]
>Yeah sorry for not knowing all 100k lines of code by heart.

It's actually 300k

**Comment by Lumipharon** [comment]
>doesn't know how many lines of code we have by heart

Scrub coder confirmed.

**Comment by Jordie0608** [comment]
This is not how you OOP, make them children of noslips.

**Comment by Robustin** [comment]
Yea my comment was a bit silly but the fact combat boots were sitting there a few lines down with identical armor values on top of all the other people going TL;DR "they already have noslips/4TC was a good price/don't give them free noslips" was also getting a bit silly.

**Comment by sirnat** [comment]
Why would the syndicate give traitors no slips who have a higher chance of death instead of their ops. 

Why not give their highly trained, more death prone soldiers shoes that have more grip so they don't slip and fall down. 

Add this mang

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 2]
```diff
@@ -16,6 +16,17 @@
 	flags = NOSLIP
 	origin_tech = "syndicate=3"
 	burn_state = -1 //Won't burn in fires
+	
+/obj/item/clothing/shoes/nukegaloshes
```
Comment (Dorsisdwarf) [inline code/modules/clothing/shoes/miscellaneous.dm]: These should really be children of regular galoshes since, you know, OOP

**Comment by MrStonedOne** [comment]
Please make a changelog,

You can add it to the PR's body rather then as a file now, see here: https://tgstation13.org/phpBB/viewtopic.php?f=5&t=5157

**Comment by duncathan** [comment]
@Robustin
Do know how to fix the code issues brought up or are you still illiterate
If it's the latter I'll help you out

**Comment by Lumipharon** [comment]
Just change it back to when they could buy normal noslips.

Someone needs to make them disguisable but that's something for a seperate pr.

**Comment by Aranclanos** [comment]
inactivity

**Pull request closed without merging**